---
title: "Evaluating the Effects of Trembling Aspen Genetics on Insect Communities"
author: "Clay Morrow and Ting-Fung Ma"
subtitle: "Stat 877 project presentation"
date: "May 2, 2019"
fontsize: 10pt
header-includes:
  - \setbeamersize{text margin left=5mm,text margin right=5mm} 
output: 
  beamer_presentation:
    theme: "Berlin"
    colortheme: "dolphin"
    fonttheme: "professionalfonts"
    slide_level: 1
---

```{r pres setup, include = FALSE}
library(knitr)
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(root.dir = "../")
# load objects created in the final report
```
```{r}
data(list = c("presentation-objects",
              "GWA-annotated-candidate-table"))
library(jpeg);library(png);library(grid);library(gridExtra)
```


# Background

## Plant-insect interactions: 

* Plants are known to influence insect community structure

* Plant Diversity is positively correlated with insect diversity.

    * effects of interspecific variation are well studied
    
    * effects of **intraspecific variation** are less well studied 
    
## *Populus tremuloides* (Trembling Aspen):
 
* the most wide-spread tree species in North America

* one of the fastest-growing tree species 

* incredible amounts of intraspecific variation

* food source for an incredible diversity of herbivores    

<!-- # Conceptual Model -->

```{r}
# Picture Here
```


# Common Garden

```{r,out.width="100%",fig.align="center"}
overhead.image <- rasterGrob(readJPEG("study-photos/WisAsp_Overhead1.JPG"))
grid.arrange(overhead.image)
```

# Research Goals

## Question: How do aspen genetics shape insect communities? 

* Herbivorous insects are known to differ by aspen clone (genet) and by aspen 
traits (size, defense chemistry, nutrition, etc.).

* Is there evidence for genetic drivers other than these physical traits?

* **Are there genomic regions that affect multiple insects?**

```{r,out.width="50%", fig.align="center"}
library(png);library(jpeg);library(grid);library(gridExtra)
image.1 <- rasterGrob(readPNG("study-photos/conceptual-model2.png"))
image.2 <- rasterGrob(readPNG("study-photos/genes-to-insects-conceptual-model.png"))
grid.arrange(image.1,nrow = 1)
```


# Genome-wide association analysis

## GWA of insect incidence on aspen common garden:

* `r insect.n` common insect species; present/absent (1/0)

* `r prettyNum(tree.n, big.mark = ",")` trees from `r genet.n` genets

* 8 tree trait covariates

* 4 time periods (longitudinal)

* `r prettyNum(SNP.N, big.mark = ",")` SNP regions

```{r, out.width="50%",fig.align="center"}
grid.arrange(image.2)
```


<!-- with 3 possible genotypes {0,1,2} -->
<!-- # of alleles that match the reference allele -->


# Statistical Methods

## Model: Univariate SNP associations

$$\text{logit}(p_{ijk_{g}}) = \beta_0 +\alpha \text{G}_{g} + x_{jk_{g}}^\intercal\beta + \varepsilon_{g(j)}$$

* $\text{logit}(y)=\log\left(\frac{y}{1-y}\right)$

* SNPs considered independently

* $p_{ijk_{g}}$: probability of observing $>0$ individual insects of species $i$
during survey event $j$ on tree $k$ of genet $g$

* $\text{G}_{g}$: SNP-specific genotype of genet $g$

* $x_{jk_{g}}$: vector of observed tree trait covariates for tree $k$ during
survey event $j$

* $\varepsilon_{g(j)}$: nested random effect of genet within survey event

<!-- # Diagnostics: p value distributions -->

<!-- ```{r,out.width="30%",out.height="45%"} -->
<!-- insects <- c("Harmandia", "Petiole-Gall", -->
<!--              "Phyllocolpa","Blotch-Mine", -->
<!--              "Ants","Green-Sawfly") -->
<!-- paths <- normalizePath(file.path("figures/p-hist",paste0("p-histogram-of_",insects,".png"))) -->
<!-- knitr::include_graphics(path = paths) -->
<!-- ``` -->

# Statistical Methods

## Computation:

* lme4::glmer() is used to fit the GLMM with the nested random effect

* High Throughput Computing (CHTC) for parallelization

* A total of 2,059,560 ($18\text{ insects}\times114,420\text{ SNPs}$) GLMMs were fit.

* Storey's q values of coefficient $\alpha$ used to select significant SNPs (.05 cutoff)

# Diagnostics: Bonferroni vs. Storey's q 

```{r, out.width = "50%", fig.align="center"}
p.man <- normalizePath(file.path("figures/p-manhattan/Manhattan-plot_Petiole-Gall.png"))
q.man <- normalizePath(file.path("figures/q-manhattan/Manhattan-plot_Petiole-Gall.png"))

img.1 <- rasterGrob(as.raster(readPNG(p.man)))
img.2 <- rasterGrob(as.raster(readPNG(q.man)))
grid.arrange(img.1,img.2,ncol = 1)
```

\scriptsize

* Using q values (bottom) allows for discovery of many more potential associations than bonferroni p values (top).

* significant q values contain all significant bonferroni p values

# Results: Significant SNPs per Insect

\scriptsize

## 

```{r}
knitr::kable(sig.n.table)
```

\normalsize 

There are `r prettyNum(snps.n, big.mark = ",")` unique significant SNPs. 


# Insects:

```{r, out.width="100%",fig.align="center"}
harm.image <- rasterGrob(readJPEG("study-photos/Harmandia-photo.png"))
pet.image <- rasterGrob(readPNG("study-photos/Petiole-gall-picture.png"))
phyl.image <- rasterGrob(readJPEG("study-photos/Phyllocolpa-picture.jpg"))
smk.1 <- rasterGrob(readJPEG("study-photos/smokey-aphid-picture.png"))
smk.ant <- rasterGrob(readJPEG("study-photos/smokey-aphid-ant-picture.png"))
grn.aph <- rasterGrob(readPNG("study-photos/green-aphid-picture.png"))


grid.arrange(harm.image,grn.aph,phyl.image,pet.image,smk.1,smk.ant,nrow = 2)
```




----

## Results: Shared Associations

```{r multiman,fig.align='center', out.width="100%",fig.align="left"}
path <- normalizePath(file.path("figures/gg-multiManhattanPlot.png"))
knitr::include_graphics(path)
```

\scriptsize
The matched genes of these `r common.n` shared association markers (red) 
make up our candidate list.

# Results: Imputed Gene Function

## Top matched genes with annotations imputed from congeneric *Populus trichocarpa*:

\tiny
```{r}
library(dplyr)
table <- candidate.table %>% select(-n.assoc) %>% head(n = 21)
small.tab.names <- c("Potri Gene","%match","Descr.","avg allele. freq", "insects")
knitr::kable(table, col.names = small.tab.names)
average.freq <- mean(candidate.table$mean_alleleFreq)
```


<!-- These annotations are vague and will require more research to determine  -->
<!-- which make biological sense. -->

<!-- "*A. t.* **glucuronate:xylan alpha-glucuronosyltransferase 1**: Glycosyltransferase required for the addition of both glucuronic acid and 4-O-methylglucuronic acid branches to xylan in stem cell walls. ... is responsible for almost all of the substitutions of the xylan backbone in stem glucuronoxylan." -->

\normalsize

# Discussion

## Conclusions

* Evidence of genomic regions that influence individual insects and communities

* These genomic regions are independent of the observed tree traits

## Future Directions 

* More thoroughly research gene annotations for *Populus trichocarpa* and model *Arabidopsis thaliana*.

* Compare with other methods (multi-SNP, BLUP association, GBlups, etc)

    * A similar study on this garden found only 2 insect associations with < 12 total significant SNPs using BLUP association and a more lenient inclusion cutoff ($q < 0.1$) in 2015
    
* Include more tree traits and environmental variables and interactions



# Extra Slides
<!-- for lab group -->
<!-- Potential Problems -->

-----

\tiny Potential Issues (Simulated Data)

```{r, out.width="100%", out.height="95%"}
n0 = n1 = round(genet.n/3)
n2 = genet.n - (n0 + n1)
n0.2 = round(.051*genet.n)
n1.2 = round(.3*genet.n); n2.2 = genet.n-(n0.2+n1.2)
set.seed(237)

toy.data <- data.frame(
  alleles.A = c(rep(0,n0),
                rep(1,n1),
                rep(2,n2)),
  Genotype = factor(c(rep("homo.B",n0),
                      rep("hetero",n1),
                      rep("homo.A",n2))),
  SNP.1  = c(rbinom(prob = .6,n = n0, size = n0)/n0,
             rbinom(prob = .5,n = n1, size = n1)/n1,
             rbinom(prob = .4,n = n2, size = n2)/n2))
  
toy.data2 <- data.frame(
  alleles.A = c(rep(0,n0.2),
                rep(1,n1.2),
                rep(2,n2.2)),
  Genotype = factor(c(rep("homo.B",n0.2),
                      rep("hetero",n1.2),
                      rep("homo.A",n2.2))),
  SNP.2  = c(
             rnorm(n = n0.2,sd = .05,mean = .7),
             # rbinom(prob = .5,n = n0.2, size = n0.2)/n0.2,
             rbinom(prob = .6,n = n1.2, size = n1.2)/n1.2,
             rbinom(prob = .7,n = n2.2, size = n2.2)/n2.2)
)

toy.data3 <- data.frame(
  alleles.A = c(rep(0,n0),
                rep(1,n1),
                rep(2,n2)),
  Genotype = factor(c(rep("homo.B",n0),
                      rep("hetero",n1),
                      rep("homo.A",n2))),
  SNP.3  = c(
             rbinom(prob = .7,n = n0, size = n0)/n0,
             rbinom(prob = .6,n = n1, size = n1)/n1,
             rbinom(prob = .7,n = n2, size = n2)/n2)
)

library(ggplot2)
g1 <- ggplot(data = toy.data, aes(x = alleles.A, 
                            y = SNP.1, col = Genotype)) +
  labs(x = "",
         y = "Probability of Insect Presence",
       title = "Association with SNP 1 \nHypothesized Situation",
       subtitle = "Equal distribution of alleles") + 
  theme(legend.position = "none") +
  geom_point() + geom_smooth(method = "lm", aes(col = NULL), col = "black") 

g2 <- ggplot(data = toy.data2, aes(x = alleles.A, 
                            y = SNP.2, col = Genotype)) +
  labs(x = "",
         y = "",
       title = "Association with SNP 2: \nHeterozygote Advantage",
       subtitle = "Uneven distribution of alleles") + 
  theme(legend.position = "none") + 
  geom_point() + geom_smooth(method = "lm", aes(col = NULL), col = "black")

g3 <- ggplot(data = toy.data3, aes(x = alleles.A, 
                            y = SNP.3, col = Genotype)) +
  labs(x = "Alleles equal to reference allele",
         y = "",
       title = "Association with SNP 3: \nHeterozygote Advantage",
       subtitle = "Equal distribution of alleles") + 
  theme(legend.position = "none") +
  geom_point() + geom_smooth(method = "lm", aes(col = NULL), col = "black")

grid.arrange(g1,
             g3,g2,nrow = 1)
```

<!-- #### -->
<!-- end for lab group -->

----

## BLUPs: Linear Mixed Model (LMM) Definition

Consider the following model for some response $Y$ on observation $j$:

$$Y_j = X\beta + Z\tau + \varepsilon $$

* $E(Y_j = X\beta)$

* $X$ and $Z$ are design matrices of independent variable values for each observation 

* $\beta$ is an unknown vector of regression parameters (fixed effects)

* $\tau$ is an unknown vector of **random effects** 

    * $E(\tau)=0$ and $Var(\tau)=G$; $G$ is a variance-covariance matrix

* $\varepsilon$ is an unknown vector of random errors

    * $E(\varepsilon)=0$ and $Var(\varepsilon)=R$

* Many approaches differ by methods of estimating $G$ and $R$.

----

## LMM: Parameter Estimation

$$Y_j = X\beta + Z\tau + \varepsilon$$

 * $\tau \sim N(0,G)$, $\varepsilon \sim N(0,R)$, and $Cov(\tau,\varepsilon)=0$

 * Find coefficients $\hat{\beta}$ and $\hat{\tau}$ such that they:

    * minimize variance prediction error $Var(Y_{j} - \hat{Y_{j}})$

    * are constrained by $E(Y_{j} - \hat{Y_{j}})=0$
    
          ($Bias(\hat{\theta})=E(\hat{\theta}) - \theta$)

 * $\hat{\beta}$ is a vector of best linear unbiased estimators (BLUEs) of fixed effects

 * $\hat{\tau}$ is a vector of best linear unbiased predictors (BLUPs) of random effects

 * conditional variances (covariance matrices for $\beta$ and $\varepsilon$) are often unknown and are
estimated as a nuisance parameter with Bayesian EM algorithm (`nlme`,`lme4`, etc.)

## Take-Home Message

BLUPs are constrained coefficienct estimates of random effects.

----

## Hilary's GWA Model (**I think**)

* First fit LMM for a trait of interest (Size) with $\beta_i$ fixed effects and $\tau_g$ random effects:

$$\text{Size} = \mu +  \beta_1(\text{PGs}) + \beta_2(\text{CTs})+ \dots + \beta_n(\text{Trait}_n) + \tau_{g}(\text{Genet}_{g}) + \varepsilon$$
 
* Then conduct a linear association analysis between the size-specific BLUPs $\tau$ and the Genotype of at SNP $s$ of each genet $g$, possibly with other trait BLUPs as covariates:

$$\tau_{g,i} = \mu +\alpha(\text{Genotype}_{s,g}) + [\tau_{g,j}]+ \varepsilon$$

```{r, out.width="80%",fig.asp=.33,fig.align="center"}
dat = data.frame(
  y = c(rnorm(n = 100, mean = 1.5,sd = .5),
        rnorm(n = 100, mean = 0, sd = .5),
        rnorm(n = 100, mean = -1.5, sd = .5)),
  x = rep(0:2, each = 100),
  Genotype = factor(rep(c("homo.B","hetero","homo.A"),each = 100))
)

ggplot(data = dat, aes(x = x, y = y, col = Genotype)) + 
  geom_point() + 
  geom_smooth(method = "lm",aes(col = NULL), col = "black") +
  labs(y = expression(tau), x = "Alleles matching reference allele") +
  theme(axis.title = element_text(face = "bold"),
        axis.title.y = element_text(size = 20))
  
```

----

## Modern GWA Models

$$ Y = \text{trait}_i\beta + \text{SNP}_j\tau $$

Where:

* SNP is a matrix of allele frequencies of each SNP for each individual

* Most often SNPs are considered individually (single-marker GWA)

However, they can be assessed together: this version has far more predictors than observation ($p > n$) which is statistically difficult.

* Usually done with shrinkage approaches (often Bayesian) to remove SNPs with 
weak effects. 

## Or

For predictive purposes, models that are re-fit with top $x$ SNP effects from
single-marker GWA can perform as well or better than multi-marker models
(Kristensen et al., 2018).
